### ISSUE TITLE
ScottPlot 5 Scatter Plot: data containing NaN and Infinity

### ISSUE THREAD

**Issue by Hub3r** [ScottPlot/ScottPlot#3276]
**Suggestion:** With the release of ScottPlot 5 it seems that a feature where "gaps" were created on encounter a double.NaN value has been removed. As described here [https://scottplot.net/faq/nan/](in the 4.1 cookbook) I am asking that this feature be re-introduced.

```cs
var plt = new ScottPlot.Plot(600, 400);
var scatter1 = plt.AddScatter(xs, ys, Color.Gray);
var scatter2 = plt.AddScatter(xs, ysWithNan, Color.Black);
scatter2.OnNaN = ScottPlot.Plottable.ScatterPlot.NanBehavior.Gap;
```

![scatter_nan_gap](https://github.com/ScottPlot/ScottPlot/assets/17992712/17b39fe8-e438-4b6a-b933-1cea849c0c02)

**Comment by jamaa** [comment]
Yes, I recently started using Scottplot (4) and definitely need this for being able to switch to SP5. In SP4, the `.OnNaN` property only exists for Scatterplots, but not for the more performant signal plots. Any chance of also supporting NaN gaps for signal plots in SP5?

**Comment by jamaa** [comment]
sounds like this has been fixed with #3577?

This FAQ page should probably be updated accordingly for SP5: https://scottplot.net/faq/nan/
